src/open_inwoner/cms/plugins/cms_plugins/tasks.py

```diff
@@ -0,0 +1,89 @@
+from django.utils import formats
+from django.utils.translation import gettext as _
+
+from cms.plugin_base import CMSPluginBase
+from cms.plugin_pool import plugin_pool
+from objectsapiclient.models import Configuration
+
+from ..models import TasksConfig
+from ..types import Object, ObjecttypeTaak
+
+
+@plugin_pool.register_plugin
+class TasksPlugin(CMSPluginBase):
+    """
+    Uses the Objects API to retrieve and show tasks according to the MijnTaken Objecttypes schema
+    Reuses the UserFeedPlugin template
+    """
+
+    model = TasksConfig
+    name = _("Task list Plugin")
+    render_template = "cms/plugins/tasks/tasks.html"
+    cache = False
+
+    def render(self, context, instance, placeholder):
+        request = context["request"]
+        context["instance"] = instance
+        context["tasks"] = []
+
+        if not request.user.is_authenticated or not (bsn := request.user.bsn):
+            return context
+
+        task_objects = self.get_tasks_by_bsn(user_bsn=bsn)
+
+        task_dicts = []
+        for task in task_objects:
+            task_data = {
+                "titel": task.titel,
+                "soort": task.soort,
+                "status": task.status,
+                "verloopdatum": formats.date_format(
+                    task.verloopdatum,
+                    format="DATETIME_FORMAT",
+                    use_l10n=True,
+                ),
+                "identificatie": task.identificatie.value,
+                "koppeling": task.koppeling,
+                "verwerker_taak_id": str(task.verwerker_taak_id),
+                "eigenaar": task.eigenaar,
+            }
+
+            match task.soort:
+                case "url":
+                    task_data["task_url"] = str(task.url.uri)
+                case "portaalformulier":
+                    task_data["task_url"] = str(task.portaalformulier.formulier.value)
+                case "ogonebetaling":
+                    task_data["ogonebetaling"] = task.ogonebetaling
+                case _:
+                    raise ValueError("Unsupported taak soort: %s", task.soort)
+
+            task_dicts.append(task_data)
+
+        context["tasks"] = task_dicts
+
+        return context
+
+    def get_tasks(self):
+        objects_api_client = Configuration.get_solo().client
+
+        objects = (
+            (Object(**obj.__dict__))
+            for obj in objects_api_client.get_objects(
+                object_type_uuid=self.model.object_type
+            )
+        )
+
+        return (ObjecttypeTaak(**obj.record.data) for obj in objects)
+
+    def get_tasks_by_bsn(self, user_bsn):
+        tasks = []
+        for task in self.get_tasks():
+            identificatie = task.identificatie
+
+            if identificatie.type != "bsn" or identificatie.value != user_bsn:
+                continue
+
+            tasks.append(task)
```

Comment:
Maybe a list comprehension is easier to read?

```suggestion
        tasks = [task for task in self.get_tasks() if task.identificatie.type == "bsn" or task.identificatie.value == user_bsn]
```